src/CTA.WebForms2Blazor/FileConverters/ProjectFileConverter.cs

```diff
@@ -1,33 +1,108 @@
+            {
+                // Since we're using default rules, we want to specify where to find those rules (once they are downloaded)
+                cli.RulesDir = Constants.RulesDefaultPath;
+            }
+            
+            var packageReferences = new Dictionary<string, Tuple<string, string>>
+            {
+                { "Autofac", new Tuple<string, string>("4.9.1.0", "4.9.3")},
+                { "EntityFramework", new Tuple<string, string>("6.0.0.0", "6.4.4")},
+                { "log4net", new Tuple<string, string>("2.0.8.0", "2.0.12")},
+                { "Microsoft.Extensions.Logging.Log4Net.AspNetCore", new Tuple<string, string>("1.0.0", "2.2.12")}
+            };
+                
+            // Create a configuration object using the CLI and other arbitrary values
+            PortCoreConfiguration projectConfiguration = new PortCoreConfiguration()
+            {
+                ProjectPath = cli.FilePath,
+                RulesDir = cli.RulesDir,
+                IsMockRun = cli.IsMockRun,
+                UseDefaultRules = cli.DefaultRules,
+                PackageReferences = packageReferences,
+                PortCode = false,
+                PortProject = true,
+                TargetVersions = new List<string> { cli.Version }
+            };
+            
+            // Instantiating a solution port object downloads the default rules files. This is otherwise not needed
+            // var solutionPort = new SolutionPort(solutionPath,
+            //     new List<AnalyzerResult> {analyzerResult},
+            //     new List<PortCoreConfiguration> {projectConfiguration});
+
+            // Use the analyzer results and project config to create a project rewriter
+            var projectRewriter = new ProjectRewriter(analyzerResult, projectConfiguration);
+
+            // Initialize() will run a rules analysis and identify which porting rules to execute.
+            // This will also detect which packages to add to the new .csproj file
+            var projectResult = projectRewriter.Initialize();
+            var projectActions = projectResult.ProjectActions;
+            var packages = projectActions.PackageActions.Distinct()
+                .ToDictionary(p => p.Name, p => p.Version);
+
+            // Now we can finally create the ProjectFileCreator and use it
+            var projectReferences = analyzerResult?.ProjectBuildResult?.ExternalReferences?.ProjectReferences.Select(p => p.AssemblyLocation).ToList();
+            var metaReferences = analyzerResult?.ProjectBuildResult?.Project?.MetadataReferences?.Select(m => m.Display).ToList();
+            var projectFileCreator = new ProjectFileCreator(FullPath, projectConfiguration.TargetVersions, packages, 
+                projectReferences, ProjectType.WebForms, metaReferences);
 
-        public override Task<IEnumerable<FileInformation>> MigrateFileAsync()
+            return projectFileCreator.CreateContents();
+        }
```

Comment:
Per our discussion, refactor out most of this since we can assume CTA will pass in most of the information we need, then move the hard coded values to a Program.cs and/or unit tests to test this class